code/datums/flock/flock.dm

```diff
@@ -478,8 +492,12 @@
 	//cleanup as necessary
 	if(src.flockmind)
 		hideAnnotations(src.flockmind)
-	for(var/mob/M in src.units)
-		hideAnnotations(M)
+	for(var/mob/living/intangible/flock/trace/T as anything in src.traces)
+		T.death()
+	for(var/mob/living/critter/flock/F as anything in src.units)
+		F.dormantize()
+	for(var/obj/flock_structure/S as anything in src.structures)
+		S.flock = null
```

Comment:
```suggestion
		src.removeStructure(S)
```
and probably add `S.flock = null` to that function. It's probably an oversight that we hadn't done that already.